src/os/bluestore/BlueStore.cc

```diff
@@ -5182,10 +5115,16 @@ void BlueStore::Collection::split_cache(
       };
 
       for (auto& e : o->extent_map.extent_map) {
-	e.blob->last_encoded_id = -1;
+        e.blob->last_encoded_id = -1;
       }
       for (auto& b : o->extent_map.spanning_blob_map) {
-	b.second->last_encoded_id = -1;
+        b.second->last_encoded_id = -1;
+      }
+
+      for (auto &i : o->bc.buffer_map) {
+        ldout(store->cct, 1)
+          << __func__ << "   moving " << i.second << dendl;
+        dest->cache->_move(cache, &i.second);
```

Comment:
You still need those buffers in the BufferSpace to keep referring to them, so I think it made sense to move them all? I don't know if I missed some edge cases here

Reply:
wait... there should be no "writing" buffers in the cache. But they're still in bc.buffer_map. So there are two issues with this code now:
1. it will wrongly remove the buffer from the source cache. Looking at a.g. LruBufferCacheShard::_rm method implementation I can see no protection from removing a buffer which doesn't belong to it.
2. it will wrongly add it to the destination one. That's evidently an inconsistent state - "writing" buffers shouldn't be there.